### ISSUE TITLE
Libs are truncated when the line is too long

### ISSUE THREAD

**Issue by Nathan-PERIER-Bpk** [pkgconf/pkgconf#368]
Hello,

We have a C++ build pipeline with a lot of dependencies to static libraries generated in a previous step. These files are stored within the build environment, so the paths are quite long. The result is a 120kB `.pc` file, most of which is dedicated to libraries.

With this (rather unorthodox) setup, we have noticed that the libs generated by `pkgconf` are incomplete. More specifically, it looks like the input line gets truncated at some point.

We have created a dummy `.pc` file ([test.pc.zip](https://github.com/user-attachments/files/17660952/test.pc.zip)) with 1200 static libraries to reproduce the issue. With this, we get the following output :

```
$ pkgconf --libs test
... /workspace/test/build_debug/staging/proj_dep-v3.5_45d2236a/lib64/libdep_cmptf11e0b0549dd2d73.a /workspace/test/build_debug/staging/proj_dep-v3.5_
```

We have noted that the last library to be printed fully is the 689th one in the list, and the 690th path is incomplete. Also, if we remove the `-L${prefix}` at the beginning of the list, we notice that the total size of the output is very close to 65535 :

```
$ pkgconf --libs test | wc -c
65529
```

(actually, we get exactly 65535 if we add the size of the `Libs: ` at the beginning of the line in the `.pc` file)

After a bit of digging, we found the `PKGCONF_BUFSIZE` variable in `libpkgconf/libpkgconf.h` and we think this is where the issue comes from.

Is this a known limitation of pkgconf ? To be fair, our use case is not exactly common...

Also, would you advise us to change the value of `PKGCONF_BUFSIZE` and compile our own binary ?

Thanks !

**Comment by kaniini** [comment]
Yes, at the moment, fragments are assembled into a 64KB buffer.  You can increase PKGCONF_BUFSIZE as a workaround, but it may be better to just implement a proper string type for this where we can do fast realloc(2) calls to extend the length of these buffers.

**Comment by kaniini** [comment]
Should be fixed by d0f8f3f2289b8618a1dfadc6c3ec30c2fbe3030d

**Comment by Nathan-PERIER-Bpk** [comment]
Oh nice ! We ended up using a version with increased buffer size but this will be much more reliable. Thank you !
